app/controllers/reg/SummaryController.scala

```diff
@@ -85,7 +85,16 @@ trait SummaryController extends AuthenticatedController with CommonService with
                   case AccountingDetailsSuccessResponse(response) => response
                   case _ => throw new Exception("could not find company accounting details")
                 }
-                Ok(Summary(details.companyName, details.jurisdiction, accountDates, ppobAddress, rOAddress, ctContactDetails, tradingDetails, takeoverDetails, cc))
+                optTakeoverDetails match {
+                  case Some(TakeoverDetails(true, None, None, None, None)) =>
+                    Redirect(controllers.takeovers.routes.TakeoverInformationNeededController.show())
+                  case Some(TakeoverDetails(true, Some(_), Some(_), Some(_), Some(_))) =>
+                    Ok(Summary(details.companyName, details.jurisdiction, accountDates, ppobAddress, rOAddress, ctContactDetails, tradingDetails, optTakeoverDetails, cc))
+                  case Some(TakeoverDetails(true, _, _, _, _)) =>
+                    Redirect(controllers.takeovers.routes.TakeoverInformationNeededController.show())
```

Comment:
If you change the order of these cases then you'd only need two cases
```suggestion
                  case Some(TakeoverDetails(true, Some(_), Some(_), Some(_), Some(_))) =>
                    Ok(Summary(details.companyName, details.jurisdiction, accountDates, ppobAddress, rOAddress, ctContactDetails, tradingDetails, optTakeoverDetails, cc))
                  case Some(TakeoverDetails(true, _, _, _, _)) =>
                    Redirect(controllers.takeovers.routes.TakeoverInformationNeededController.show())
```